compiler/modules/nback/nback.bal

```diff
@@ -6,11 +6,13 @@ import wso2/nballerina.nback.llvm;
 
 # The preferred output extension for the output filename.
 public const OUTPUT_EXTENSION = ".ll";
+configurable string target = "x86_64-pc-linux-gnu";
 
 // If outputFilename is non-nil write the output to this file.
 public function compileModule(bir:Module mod, string? outputFilename) returns err:Any|io:Error? {
     llvm:Context context = new;
     llvm:Module llMod = check buildModule(mod, context);
+    llMod.setTarget(<llvm:TargetTriple>target);
```

Comment:
This isn't OK because the user will get a panic if they've set an unacceptable value for target.

Two possible approaches:
1. Make the type of `target` be `llvm:TargetTriple`: the configurable implementation should take care of the reporting an error if the user supplies an invalid string. If the configurable implementation does not handle this, then please file an issue with nBal needs.
2. If (1) does not work than have two variables: a configurable variable of type string and another variable of type llvm:TargetTriple with an initializer that uses `check` to initialize from the configurable variable.

Reply:
I created a issue for 1 (link:https://github.com/ballerina-platform/ballerina-lang/issues/31447) but I don't have permission to set labels for issues.